src/streaming/controllers/CmcdController.js

```diff
@@ -177,6 +177,11 @@ function CmcdController() {
         
         eventModeTargets.forEach(targetSettings => {
             if (targetSettings.enabled) {
+
+                if (targetSettings.events.length === 0) {
```

Comment:
Maybe check if targetSettings.events is not null

```suggestion
                if (targetSettings.events?.length === 0) {
```